linode/domainrecord/resource.go

```diff
@@ -203,3 +212,27 @@ func domainRecordTargetSuppressor(k, provisioned, declared string, d *schema.Res
 	return len(strings.Split(declared, ".")) == 1 &&
 		strings.Contains(provisioned, declared)
 }
+
+// reconcileName handles cases where the user has specified their FQDN as a part of their
+// planned record name but the API trims the FQDN from the returned record name.
+func reconcileName(ctx context.Context, client linodego.Client, d *schema.ResourceData, apiName string) (string, error) {
```

Comment:
I originally wanted to implement this as a DiffSuppressFunc, but couldn't because the ProviderMeta (and by extension the API client) is not accessible from within DiffSuppressFuncs.

I also tried making a CustomizeDiff function for this same purpose, but Terraform would always report an inconsistent plan after the diff was dropped on the `name` field.

This implementation was inspired by the [LKE matchPoolsWithSchema function](https://github.com/linode/terraform-provider-linode/blob/dev/linode/lke/cluster.go#L347) and seems to be the most reliable of the three solutions. 

I was initially concerned that this is unintended behavior, but the [SDKv2 docs](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/data-consistency-errors) imply that providers should always "always set an attribute state value to the exact configuration value or prior state value".